Mathlib/Probability/Kernel/Irreducible.lean

```diff
@@ -0,0 +1,106 @@
+/-
+Copyright (c) 2025 Bryan Wang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bryan Wang
+-/
+module
+
+public import Mathlib.Probability.Kernel.Composition.Comp
+
+/-!
+# Irreducibility of kernels
+
+A kernel `κ : Kernel α α` is `φ`-irreducible, for a given measure `φ` on `α`,
+if for every measurable set `A` with positive measure under `φ`,
+there exists a positive integer `n` such that for every `a : α`, we have `(κ ^ n) a A > 0`.
+
+When the kernel `κ` is the transition kernel of a Markov chain,
+this precisely means that the Markov chain is `φ`-irreducible,
+that is, there is a positive probability of reaching any
+(`φ`-positive measure) set of states from any other state within a finite number of steps.
+
+## Main definitions
+
+* `ProbabilityTheory.Kernel.IsIrreducible`:
+  irreducibility of a given kernel with respect to a measure `φ`.
+* `ProbabilityTheory.Kernel.IsStronglyIrreducible`:
+  strong irreducibility of a given kernel with respect to a measure `φ`.
+
+## Main statements
+
+* `isIrreducible_of_le_measure`: If a kernel `κ` is irreducible with respect to a measure `φ₂`,
+  then it is also irreducible with respect to any measure `φ₁` with `φ₁ ≤ φ₂`.
+* `isStronglyIrreducible_of_le_measure`: If a kernel `κ` is strongly irreducible
+  with respect to a measure `φ₂`,
+  then it is also strongly irreducible with respect to any measure `φ₁` with `φ₁ ≤ φ₂`.
+
+## References
+
+* [Meyn, S.P. and Tweedie, R.L., *Markov Chains and Stochastic Stability*][meyntweedie1993]
+* [C Robert, G Casella, *Monte Carlo Statistical Methods*][robertcasella2004]
+
+-/
+
+@[expose] public section
+
+open MeasureTheory
+
+open scoped MeasureTheory ENNReal ProbabilityTheory
+
+namespace ProbabilityTheory
+
+variable {α β : Type*} {mα : MeasurableSpace α} {mβ : MeasurableSpace β}
+
+namespace Kernel
+
+/-- `φ`-irreducibility of a kernel `κ` (w.r.t. a measure `φ`).
+Ref. *Meyn-Tweedie* Proposition 4.2.1(ii), page 89 -/
+@[mk_iff]
+class IsIrreducible (φ : Measure α) (κ : Kernel α α) : Prop where
+  irreducible ⦃A⦄ (hA : MeasurableSet A) (hφA : φ A > 0) a :
+    ∃ (n : ℕ), (κ ^ n) a A > 0
+
+/-- Strongly `φ`-irreducibility of a kernel `κ` (w.r.t. a measure `φ`).
+Ref. https://www.stats.ox.ac.uk/~rebeschi/teaching/AdvSim/18/notes/notes4.pdf, Definition 3.1 -/
```

Comment:
That's not a proper reference. If it's not in Meyn-Tweedie or one other central reference of the field, we shouldn't probably define it unless it turns out to be extremely useful for formalization reasons.